File: client/src/api/glyph.ts
import {
  Glyph,
  GlyphCreate,
  GlyphDelete,
  GlyphEdit,
  GlyphRead,
  apiClient,
} from './core';

export const createGlyph = async (glyph: GlyphCreate) =>
  await apiClient.post<Glyph>(
    `${process.env.NEXT_PUBLIC_SERVER_URL}/glyphs/`,
    glyph
  );

export const readGlyph = async (glyph: GlyphRead) =>
  await apiClient.get<Glyph>(
    `${process.env.NEXT_PUBLIC_SERVER_URL}/glyphs/${glyph.id}`
  );

export const readAllGlyphs = async () =>
  await apiClient.get<Glyph[]>(
    `${process.env.NEXT_PUBLIC_SERVER_URL}/glyphs/list`
  );

export const readRelativeAllGlyphs = async (glyph: GlyphRead) =>
  await apiClient.get<Glyph[]>(
    `${process.env.NEXT_PUBLIC_SERVER_URL}/glyphs/list/${glyph.id}`
  );

export const editGlyph = async (glyph: GlyphEdit) =>
  await apiClient.put<Glyph>(
    `${process.env.NEXT_PUBLIC_SERVER_URL}/glyphs/`,
    glyph
  );

export const deleteGlyph = async (glyph: GlyphDelete) =>
  await apiClient.delete<Glyph>(
    `${process.env.NEXT_PUBLIC_SERVER_URL}/glyphs/${glyph.id}`
  );

File: server/internal/presentation/router/glyph.go
package router

import (
	"github.com/Doer-org/glyph/internal/infrastructure/database"
	"github.com/Doer-org/glyph/internal/infrastructure/persistance"
	"github.com/Doer-org/glyph/internal/presentation/http/handler"
	"github.com/Doer-org/glyph/internal/usecase"
)

func (r Router) InitGlyphRouter(conn *database.Conn) {
	repo := persistance.NewGlyphRepository(conn)
	uc := usecase.NewGlyphUsecase(repo)
	h := handler.NewGlyphHandler(uc)

	g := r.Engine.Group("/glyphs")
	g.POST("/", h.CreateGlyph)
	g.GET("/:id", h.ReadGlyph)
	g.GET("/user/:author_id", h.ReadGlyphsbyUserId)
	g.GET("/list", h.ReadAllGlyphs)
	g.GET("/list/:id", h.ReadRelativeAllGlyphs)
	g.PUT("/:id", h.EditGlyph)
	g.DELETE("/:id", h.DeleteGlyph)
}
